src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -867,366 +855,202 @@ void SetUpZoneSizingArrays(EnergyPlusData &state)
         thisDesDayWeather.HumRat = 0.0;
```

Comment:
So, you are doing three loops here because each of those initialization assignments is actually implemented as a loop. It's faster to do:

```cpp
for (int ts = 1; ts <= state->dataGlobal->NumOfTimeStepInHour * 24; ++ts)
   thisDesDayWeather.Temp[ts] = thisDesDayWeather.HumRat[ts] = thisDesDayWeather.Press[ts] = 0.0;
```

Reply:
I thought we weren't supposed to stack multiple equalities on the same line, for readability I guess. I'd rather do this (because I don't know any better) but did not do that here.

    thisDesDayWeather.Temp.dimension(state.dataGlobal->NumOfTimeStepInHour * 24, 0.0);

    struct DesDayWeathData
    {
        // Members
        std::string DateString; // date of design day weather values
        Array1D<Real64> Temp;   // design day temperatures at the major time step
        Array1D<Real64> HumRat; // design day humidity ratios at the major time step
        Array1D<Real64> Press;  // design day barometric pressure at the major time step
    };

    EPVector<DataSizing::DesDayWeathData> DesDayWeath;            // design day weather saved at major time step

Reply:
That was just shorthand for the comment. You can do this too. It's the same thing.

```cpp
for (int ts = 1; ts <= state->dataGlobal->NumOfTimeStepInHour * 24; ++ts) {
   thisDesDayWeather.Temp[ts] = 0.0; 
   thisDesDayWeather.HumRat[ts] = 0.0;
   thisDesDayWeather.Press[ts] = 0.0;
}
```

Reply:
> I thought we weren't supposed to stack multiple equalities on the same line, for readability I guess. I'd rather do this (because I don't know any better) but did not do that here.
> 
> ```
> thisDesDayWeather.Temp.dimension(state.dataGlobal->NumOfTimeStepInHour * 24, 0.0);
> ```

You mean you would rather do `dimension` than do `= 0.0`? Or you'd rather do `dimension` than an explicit loop? There is no difference between `dimension`, `= 0.0`, and an explicit loop when you are only talking about a single array. When you are talking about N arrays, the explicit loop is faster because you have only one loop rather than N. 

Related question, should these three arrays really be a single array of structs?

Reply:
I mean `dimension` is more compact (1 line). Whether `dimension` versus loop is better is above my pay grade. From the N arrays comment I suspect the loop is faster.

should these three arrays really be a single array of structs? I thought we were headed in that direction. Grouping like variables/arrays into a single struct. Or maybe I am not fully understanding the question.

Reply:
If `dimension` vs. loop is above your pay grade then you are underpaid. `dimension` essentially does a loop with an assignment inside. Doing one loop with one assignment inside is equivalent to `dimension`. It's also equivalent to `= 0.0` because the scalar-to-array assignment operator also expands to a loop. However, doing one loop with N assignments inside is faster than doing N loops with one assignment inside each (which is what N `dimension` or N assignment operators would do) because you are repeating the loop machinery N times. 

You understood the question perfectly. Yes, I think we are headed in that direction also.